cmd/minikube/cmd/config/profile_list.go

```diff
@@ -56,98 +58,111 @@ var profileListCmd = &cobra.Command{
 	},
 }
 
-var printProfilesTable = func() {
-
-	var validData [][]string
-	table := tablewriter.NewWriter(os.Stdout)
-	table.SetHeader([]string{"Profile", "VM Driver", "Runtime", "IP", "Port", "Version", "Status"})
-	table.SetAutoFormatHeaders(false)
-	table.SetBorders(tablewriter.Border{Left: true, Top: true, Right: true, Bottom: true})
-	table.SetCenterSeparator("|")
+func printProfilesTable() {
 	validProfiles, invalidProfiles, err := config.ListProfiles()
 
-	if len(validProfiles) == 0 || err != nil {
+	if len(validProfiles) == 0 {
 		exit.UsageT("No minikube profile was found. You can create one using `minikube start`.")
 	}
+
+	if err != nil {
+		glog.Warningf("error loading profiles: %v", err)
+	}
+
+	updateProfilesStatus(validProfiles)
+	renderProfilesTable(profilesToTableData(validProfiles))
+	warnInvalidProfiles(invalidProfiles)
+}
+
+func updateProfilesStatus(profiles []*config.Profile) {
 	api, err := machine.NewAPIClient()
 	if err != nil {
 		glog.Errorf("failed to get machine api client %v", err)
 	}
 	defer api.Close()
 
-	for _, p := range validProfiles {
-		cp, err := config.PrimaryControlPlane(p.Config)
-		if err != nil {
-			exit.WithError("error getting primary control plane", err)
-		}
-		p.Status, err = machine.Status(api, driver.MachineName(*p.Config, cp))
-		if err != nil {
-			glog.Warningf("error getting host status for %s: %v", p.Name, err)
-		}
-		validData = append(validData, []string{p.Name, p.Config.Driver, p.Config.KubernetesConfig.ContainerRuntime, cp.IP, strconv.Itoa(cp.Port), p.Config.KubernetesConfig.KubernetesVersion, p.Status})
+	for _, p := range profiles {
+		p.Status = profileStatus(p, api)
 	}
+}
 
-	table.AppendBulk(validData)
-	table.Render()
-
-	if invalidProfiles != nil {
-		out.WarningT("Found {{.number}} invalid profile(s) ! ", out.V{"number": len(invalidProfiles)})
-		for _, p := range invalidProfiles {
-			out.ErrT(out.Empty, "\t "+p.Name)
-		}
-		out.ErrT(out.Tip, "You can delete them using the following command(s): ")
-		for _, p := range invalidProfiles {
-			out.Err(fmt.Sprintf("\t $ minikube delete -p %s \n", p.Name))
-		}
+func profileStatus(p *config.Profile, api libmachine.API) string {
+	cp, err := config.PrimaryControlPlane(p.Config)
+	if err != nil {
+		exit.WithError("error getting primary control plane", err)
+	}
 
+	host, err := machine.LoadHost(api, driver.MachineName(*p.Config, cp))
+	if err != nil {
+		return ""
 	}
 
+	cr, err := machine.CommandRunner(host)
 	if err != nil {
-		glog.Warningf("error loading profiles: %v", err)
+		return ""
 	}
 
-}
+	hostname, _, port, err := driver.ControlPlaneEndpoint(p.Config, &cp, host.DriverName)
+	if err != nil {
```

Comment:
I'm not sure if this is the best approach.
In case of error, I'd prefer to log what errored.

maybe even return "unknown" :man_shrugging:

Reply:
Fixed. 
Logging and returning Unknown now.

Reply:
Thank you for the comments. Let me know for any improvements! (: